drivers/gpu/drm/vc4/vc4_vec.c

```diff
@@ -75,6 +76,27 @@
 #define VEC_SOFT_RESET			0x10c
 #define VEC_CLMP0_START			0x144
 #define VEC_CLMP0_END			0x148
+
+/*
+ * These set the color subcarrier frequency
+ * if VEC_CONFIG1_CUSTOM_FREQ is enabled.
+ *
+ * VEC_FREQ1_0 contains the most significant 16-bit half-word,
+ * VEC_FREQ3_2 contains the least significant 16-bit half-word.
+ * 0x80000000 seems to be equivalent to the pixel clock
+ * (which itself is the VEC clock divided by 8).
+ *
+ * Reference values (with the default pixel clock of 13.5 MHz):
+ *
+ * NTSC  (3579545.[45] Hz)     - 0x21F07C1F
+ * PAL   (4433618.75 Hz)       - 0x2A098ACB
+ * PAL-M (3575611.[888111] Hz) - 0x21E6EFE3
+ * PAL-N (3582056.25 Hz)       - 0x21F69446
+ *
```

Comment:
Datasheet says colour subcarrier frequency = 13.5MHz * FREQ / 2^31.

However the firmware code for setting up PAL_M (the only option which uses a custom subcarrier) is 0x223b61d1, which differs from yours. Computing the values, I agree with your numbers though.

Reply:
`0x223b61d1`, which translates to 3610402.169 Hz, is a long-running bug in the firmware. I'm not aware of a TV norm that would use this frequency ever existing. `sdtv_mode=3` indeed configures that frequency (with `VEC_CONFIG0_STD_MASK` set to 0 for NTSC), but that mode is pretty much unusable. [Here's how `sdtv_mode=3` looks on my Fushicai USBTV007 grabber](https://gist.github.com/kFYatek/0b2dc2f85ecc933b34f9df2559390dd4). See also https://github.com/raspberrypi/firmware/issues/756, there are pictures of @Sobakin76's monitors showing the same thing there.